### PR TITLE
add edit transaction test cases

### PR DESCRIPTION
**Pull request by ShroukMohamed16**

### CODE REVIEW

File: src/main/kotlin/test/TestCases.kt [hunk 1]
```diff
@@ -2,8 +2,78 @@ package test
...
+
+    //region Edit Transaction Test Cases
+
+     check(
+        testName = "When all inputs are valid and transaction exists, then return true",
+        result = false,
+        acceptedResult = false
+    )
+    check(
+        testName = "when transaction type is expenses and amount more than user balance return false",
```
Comment (mohamedd-hassan) [inline src/main/kotlin/test/TestCases.kt]: the user will have no balance as far as i know so this test case is not needed

File: src/main/kotlin/test/TestCases.kt [hunk 2]
```diff
@@ -2,8 +2,78 @@ package test
...
+        result = true,
+        acceptedResult = true
+    )
+    check(
+        testName = "when transaction type is empty should return false",
+        result = false,
+        acceptedResult = false
+    )
+    check(
+        testName = "when transaction type not supported should return false",
```
Comment (mohamedd-hassan) [inline src/main/kotlin/test/TestCases.kt]: transaction type will be likely handled through an enum so there is probably no issues that need testing on that field

File: src/main/kotlin/test/TestCases.kt [hunk 3]
```diff
@@ -2,8 +2,78 @@ package test
...
+
 //region Transactions Test Cases
-//todo: write all test cases that related with transactions here :)
+
+    //region Edit Transaction Test Cases
+
+     check(
+        testName = "When all inputs are valid and transaction exists, then return true",
+        result = false,
+        acceptedResult = false
```
Comment (mohamedd-hassan) [inline src/main/kotlin/test/TestCases.kt]: should return true but returns false

**Review by mohamedd-hassan — CHANGES_REQUESTED** [review]
All in all great work but check some of the requested changes and then it will be ready

File: src/main/kotlin/test/TestCases.kt (hunk 2 above)
Comment (ShroukMohamed16) [reply src/main/kotlin/test/TestCases.kt]: if user enters a type other than income or expenses

Comment (mohamedd-hassan) [reply src/main/kotlin/test/TestCases.kt]: i thought it would be a list they choose from, in that case this is okay

**Review by mohamedd-hassan — APPROVED** [review]
nice work

File: src/main/kotlin/test/TestCases.kt [hunk 4]
```diff
@@ -36,10 +44,11 @@ fun main(){
...
     )
+    // set today date
     check(
-        testName = "when date is empty should return false",
-        result = false,
-        acceptedResult = false
+        testName = "when date is empty should return true",
+        result = true,
+        acceptedResult = true
     )
```
Comment (MohammadAshrf) [inline src/main/kotlin/test/TestCases.kt:52]: Unclear logic in test: "when date is empty should return true" — it should return false or ture?

**Review by MohammadAshrf — CHANGES_REQUESTED** [review]
the feature branch should be named as **feature/edit_transaction_test_cases** instead of feature_test_cases/edit_transaction_test_cases.

we follow a structure where all feature branches go under the feature/ folder directly.

**Review by MohammadAshrf — COMMENTED** [review]
Overall, tests are well-structured and cover important edge cases. just a few naming and logic clarifications needed.

File: src/main/kotlin/test/TestCases.kt (hunk 4 above)
Comment (ShroukMohamed16) [reply src/main/kotlin/test/TestCases.kt:52]: If date is empty , I set today's date in date field
